src/types.rs

```diff
@@ -163,13 +169,17 @@ impl Point {
     ///
     /// assert!(dist < 1e-1)
     /// ```
-    pub fn distance_to(&self, point: &Point) -> f64 {
-        ((self.x() - point.x()).powi(2) + (self.y() - point.y()).powi(2)).sqrt()
+    pub fn distance_to(&self, point: &Point<T>) -> f64 {
```

Comment:
It also means you can remove the explicit `Copy` trait bounds specifying everywhere since `Float` is a subtype of `Copy`.